### ISSUE TITLE
phpcbf adds long conditional end comments without leading space

### ISSUE THREAD

**Issue by JDGrimes** [WordPress-Coding-Standards/WordPress-Coding-Standards#702]
Given the code:

``` php
if ( $something ) {
    /* long codes */
}
```

The expected fix is:

``` php
if ( $something ) {
    /* long codes */
} // End if().
```

The actual result:

``` php
if ( $something ) {
    /* long codes */
}// End if().
```

Introduced in #653. Changing the `commentFormat` to include a leading space appears to have no effect. The sniff being used is `Squiz.Commenting.LongConditionClosingComment`, and it is designed to not include any space before the comment. We actually have to exclude an error that it produces when space is present. I'm not sure that we can fix this without getting the upstream sniff modified.

**Comment by JDGrimes** [comment]
> Changing the commentFormat to include a leading space appears to have no effect.

This is because [`PHP_CodeSniffer::setSniffProprety()` `trim()`s the value](https://github.com/squizlabs/PHP_CodeSniffer/blob/01ee3606e57746e496fd27ef5a83ce87128b547d/CodeSniffer.php#L1538).

**Comment by JDGrimes** [comment]
Note that even manually modifying the upstream sniff to use a format with a leading space will not work correctly, because the sniff compares the comment token (without any leading whitespace tokens) to the format, and the two will not match up, causing the fixer to loop until it has run 50 times, and replaced the comment 50 times, with 50 spaces having been added in front of it as a result. 

So the upstream sniff would have to be modified to support leading whitespace.

A possible alternative to that would be to create a separate sniff that would fix the incorrect spacing of the comment.

**Comment by jrfnl** [comment]
It's only the fixing which is an issue though isn't it ? If it finds an end comment with a space before it, it will not trigger an error nor incorrectly fix it AFAICS.

Considering the configurability of this sniff in the first place is [my doing](https://github.com/squizlabs/PHP_CodeSniffer/pull/1074), I would venture a guess that an extra config property for the number of expected whitespaces before the end comment might well be considered for merge upstream.

**Comment by JDGrimes** [comment]
> It's only the fixing which is an issue though isn't it ? If it finds an end comment with a space before it, it will not trigger an error nor incorrectly fix it AFAICS.

Right.



> Considering the configurability of this sniff in the first place is my doing, I would venture a guess that an extra config property for the number of expected whitespaces before the end comment might well be considered for merge upstream.

After more thought, I've decided that I'm not a big fan of these generically patterned comments anyway. I'd rather hand-craft the comments to give the sense of what the condition is for, or what the loop is. So I won't be using patterned comments, and therefore I probably won't take it upon myself to pursue this further (though other interested parties are welcome to do so).

For the record, what I'm doing in my custom ruleset is this:

```xml
	<rule ref="WordPress-Core">
		<exclude name="Squiz.Commenting.LongConditionClosingComment.Invalid" />
	</rule>
```

Which allows me to continue getting the reports of missing comments, while ignoring the checks for "invalid" comments that don't follow the prescribed pattern.

**Comment by jrfnl** [comment]
> I'm not a big fan of these generically patterned comments anyway

@JDGrimes Nor me, but it's the only way to check the handbook rule which covers this. I think your code snippet would be a great addition to the wiki as I suspect there will be more people who would like the notification, but not the patterned comments.

**Comment by jrfnl** [comment]
Closing as the related sniff has been removed - #965
